providers/cloudflare/cloudflareProvider.go

```diff
@@ -300,3 +305,17 @@ func getProxyMetadata(r *models.RecordConfig) map[string]string {
 		"proxy": fmt.Sprint(proxied),
 	}
 }
+
+func (c *CloudflareApi) EnsureDomainExists(domain string) error {
+	err := c.fetchDomainList()
```

Comment:
If we fetch the domain list when we create the provider, lets just store it and not need to fetch it all over the place again.